sdk/eventhub/event-hubs/samples/websockets.ts

```diff
@@ -16,8 +16,8 @@
 
 import { EventHubClient } from "@azure/event-hubs";
 import WebSocket from "ws";
-const url = require("url");
-const httpsProxyAgent = require("https-proxy-agent");
+import url from "url";
+import httpsProxyAgent, { HttpsProxyAgentOptions } from "https-proxy-agent";
```

Comment:
shouldn't this be 

`import { httpsProxyAgent, HttpsProxyAgentOptions } from "https-proxy-agent"`

?

Reply:
It actually can't be neither `import httpsProxyAgent from "https-proxy-agent";` or `import { HttpsProxyAgent } from "https-proxy-agent";`. Seems like I need to add the current export as the default property of the exported value. I'll elaborate:

Right now, https-proxy-agent is exporting it's default method by just doing:

https://github.com/TooTallNate/node-https-proxy-agent/blob/master/index.js#L16

I'm guessing it has to be changed to something like:

```javascript
module.exports = HttpsProxyAgent;
module.exports.default = module.exports;
```

Would that work? Seems risky.

I'll change these imports to use `require` for now.

Reply:
In that I case, I would skip the import and use of `HttpsProxyAgentOptions` altogether.

Reply:
The issue is that this code used to fail using `tsc` 🤔 that's why I opened https://github.com/Azure/azure-sdk-for-js/issues/3064

Reply:
Its hard to believe that the lack of typing for the `options` object was what caused `tsc` to fail

It feels weird to import/require from the same module twice

Can't we do `import httpsProxyAgent from "https-proxy-agent"` and then not use the type when defining `options`?

Reply:
I'll give it a try again! I'll come back with updates